src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2695,7 +2695,9 @@ trait Types
   private final class ClassArgsTypeRef(pre: Type, sym: Symbol, args: List[Type]) extends ArgsTypeRef(pre, sym, args)
   private final class AliasNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym) with AliasTypeRef
   private final class AbstractNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym) with AbstractTypeRef
-  private final class ClassNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym)
+  private final class ClassNoArgsTypeRef(pre: Type, sym: Symbol) extends NoArgsTypeRef(pre, sym){
+    override def contains(sym: Symbol) = this.sym == sym || super.contains(sym)
```

Comment:
I'd replace `super.contains(sym)` with `pre.contains(sym)`. Sorry for not being more explicit last time.

Reply:
After that, good to go!

Reply:
Ok. Since we are at it, can we apply this trick for other classes too?

Reply:
It seems to me that specializing the `contains` methods (and maybe `find`) too, so as to avoid creating the `ContainsCollector` can be done in a different PR.

Reply:
Good question! For example, it would not work for e.g. an alias type ref, because `contains` also checks the right hand side of the alias (by `normalize`ing the type). That makes me wonder, actually, if my previous comment was correct. If the `ClassNoArgsTypeRef` refers to a polymorphic class, `normalize` would turn it into a `PolyType`, but we're ok with our shortcut, since `contains` should not match the type params (because they are local -- there should be no way an outside symbol can be equal), nor their info (a type param's bound could potentially contain a type we're looking for, but we ignore bounds in TypeCollector as far as I can tell). 😅

Reply:
In any case, yes, let's defer further tweaks to another PR -- this one is more than polished enough to merge.